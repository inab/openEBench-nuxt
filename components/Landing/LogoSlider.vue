<template>
	<v-container>
		<h2 class="text-h4">Our partners</h2>
		<v-slide-group
			ref="slideGroup"
			v-observe-visibility="visibilityChanged"
			class="pa-4"
			show-arrows="always"
		>
			<v-slide-item
				v-for="(image, index) in shuffledImages"
				:key="`logoslider_` + index"
				disabled
			>
				<a
					:href="image.href"
					target="_blank"
					class="d-flex"
					:aria-label="image.title"
				>
					<v-img
						contain
						class="ma-5 pa-2"
						max-height="120"
						max-width="200"
						width="200"
						:src="image.src"
					>
					</v-img>
				</a>
			</v-slide-item>
		</v-slide-group>
	</v-container>
</template>

<script>
import { ObserveVisibility } from 'vue-observe-visibility';

export default {
	name: 'LogoSlider',
	directives: {
		ObserveVisibility,
	},
	data() {
		return {
			images: [
				{
					title: 'Elixir',
					href: 'https://elixir-europe.org/',
					src: require('~/static/images/partners/elixir.png'),
				},
				{
					title: 'Eucan Image',
					href: 'https://eucanimage.eu/',
					src: require('~/static/images/partners/eucanimage.svg'),
				},
				{
					title: 'EUCANcan',
					href: 'https://eucancan.com/',
					src: require('~/static/images/partners/eucancan.png'),
				},
				{
					title: 'PerMedCoE',
					href: 'https://permedcoe.eu/',
					src: require('~/static/images/partners/permedcoe.png'),
				},
				{
					title: 'IPC',
					href: 'https://ipc-project.eu/',
					src: require('~/static/images/partners/ipc.png'),
				},
				{
					title: 'decider',
					href: 'https://www.deciderproject.eu/',
					src: require('~/static/images/partners/decider.jpg'),
				},
			],
		};
	},
	computed: {
		shuffledImages() {
			// eslint-disable-next-line vue/no-side-effects-in-computed-properties
			return this.images.sort(() => Math.random() - 0.5);
		},
	},
	methods: {
		visibilityChanged(isVisible) {
			// workaround for a bug https://github.com/vuetifyjs/vuetify/issues/14195
			if (isVisible) {
				setTimeout(
					function () {
						this.$refs.slideGroup.setWidths();
					}.bind(this),
					500
				);
			}
		},
	},
};
</script>
