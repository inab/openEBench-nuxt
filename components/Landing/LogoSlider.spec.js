import { mount } from '@vue/test-utils';
import LogoSlider from './LogoSlider.vue';

const factory = () => {
	return mount(LogoSlider, {
		...createComponentMocks({}),
		mocks: {
			$vuetify: { breakpoint: {} },
		},
	});
};

describe('LogoSlider', () => {
	it('is instantiated', () => {
		const wrapper = factory();
		expect(wrapper).toBeTruthy();
	});
});
