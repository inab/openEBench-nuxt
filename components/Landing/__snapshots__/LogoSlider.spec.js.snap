// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoSlider should match snapshot 1`] = `
"<div class=\\"container\\">
  <h2 class=\\"text-h4\\">Our partners</h2>
  <div class=\\"pa-4 v-item-group theme--light v-slide-group\\">
    <div class=\\"v-slide-group__prev v-slide-group__prev--disabled\\">
      <transition-stub name=\\"fade-transition\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-icon--disabled mdi mdi-chevron-left theme--light\\"></i></transition-stub>
    </div>
    <div class=\\"v-slide-group__wrapper\\">
      <div class=\\"v-slide-group__content\\">
        <div class=\\"v-image v-responsive ma-5 pa-2 theme--light\\" style=\\"max-height: 150px; max-width: 200px;\\">
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-image v-responsive ma-5 pa-2 theme--light\\" style=\\"max-height: 150px; max-width: 200px;\\">
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-image v-responsive ma-5 pa-2 theme--light\\" style=\\"max-height: 150px; max-width: 200px;\\">
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-image v-responsive ma-5 pa-2 theme--light\\" style=\\"max-height: 150px; max-width: 200px;\\">
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-image v-responsive ma-5 pa-2 theme--light\\" style=\\"max-height: 150px; max-width: 200px;\\">
          <div class=\\"v-responsive__content\\"></div>
        </div>
        <div class=\\"v-image v-responsive ma-5 pa-2 theme--light\\" style=\\"max-height: 150px; max-width: 200px;\\">
          <div class=\\"v-responsive__content\\"></div>
        </div>
      </div>
    </div>
    <div class=\\"v-slide-group__next v-slide-group__next--disabled\\">
      <transition-stub name=\\"fade-transition\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate v-icon--disabled mdi mdi-chevron-right theme--light\\"></i></transition-stub>
    </div>
  </div>
</div>"
`;
